### PR TITLE
Cleaner mode picking UI

### PR DESCRIPTION
**Pull request by MatthewBeaudouinLafon**
Current canvas mode is highlighted (reverse video) in mode picker.
Use TAB and Shift TAB to cycle through modes in mode picker. ENTER to commit to a mode.

### CODE REVIEW

File: src/fe_modes.c [hunk 1]
```diff
@@ -122,6 +122,25 @@ void switch_mode(Mode_ID new_mode, State *state) {
...
+  if (mode < mode_first || mode_list_end <= mode) {
+    logd("mode \"%s\" is not a canvas mode", mode);
+    return mode;
+  }
+
+  // loop properly between mode_first and last mode
+  return ((mode - mode_first) + n) % (mode_list_end - mode_first) + mode_first;
+}
+
+Mode_ID next_canvas_mode(Mode_ID mode) { return add_mod_canvas_mode(mode, 1); }
```
Comment (MatthewBeaudouinLafon) [inline src/fe_modes.c:138]: Linter makes this look dumb with the one below, but not worth changing?

Comment (newsch) [reply src/fe_modes.c:138]: I looked into this a little more, the option is `AllowShortFunctionsOnASingleLine`. I'll make another pull request to change it.

**Comment by newsch** [comment]
This looks pretty slick! Is there anything else you're thinking of doing here?

**Comment by MatthewBeaudouinLafon** [comment]
> This looks pretty slick! Is there anything else you're thinking of doing here?

Nope, that about covers it. If I think of anything else, I'll make a different pull request.

**Review by newsch — APPROVED** [review]

**Pull request merged**